rust/ares/src/noun.rs

```diff
@@ -464,6 +464,24 @@ impl IndirectAtom {
         UBig::from_le_bytes_stack(stack, self.as_bytes())
     }
 
+    pub unsafe fn as_u64(self) -> Result<u64> {
+        if self.size() == 1 {
+            Ok(*(self.data_pointer()))
+        } else {
+            Err(Error::NotRepresentable)
+        }
+    }
+
+    pub unsafe fn as_u128_pair(self) -> Result<[u64; 2]> {
+        if self.size() <= 2 {
+            let u128_array = &mut [0u64; 2];
+            u128_array.copy_from_slice(&(self.as_slice()[0..2]));
+            Ok(*u128_array)
+        } else {
+            Err(Error::NotRepresentable)
+        }
+    }
```

Comment:
Rust has a native u128 type: https://doc.rust-lang.org/std/primitive.u128.html

You could make a 16-element byte array, copy into it from `atom.as_bytes()`, and then call u128::from_le_bytes() on the array to get the atom.
```suggestion
    pub unsafe fn as_u128(self) -> Result<u128> {
        if self.size() <= 2 {
            let mut u128_array = [0u8; 16];
            let self_bytes = self.as_bytes();
            u128_array[0..self.as_bytes()].copy_from_slice(self_bytes);
            Ok(u128::from_le_bytes(u128_array)
        } else {
            Err(Error::NotRepresentable)
        }
    }
```

Reply:
It's a pair of u64 to match the SoftFloat interface.

Reply:
It's not a generic u128, it's to solve a particular use case from urcrypt.

Reply:
~~urcrypt~~ urfloat, whatever we end up calling it